File: app/src/at-tamle/java/asiantech/internship/winter/ex9/binary.java
package asiantech.internship.winter.ex9;

import java.util.Scanner;
//liet ke xau nhi phan do dai n
public class binary {
    public static int process() {
        Scanner input = new Scanner(System.in);
        boolean check = false;
        int n = 0;
        while (!check) {
            System.out.println("Nhap n ");
            try {
                n = input.nextInt();
                check = true;
            } catch (Exception e) {
                System.out.println("Ban phai nhap so! hay nhap lai...");
                input.nextLine();
            }
        }
        return (n);
    }
    public static void main(String[] args) {
        int n = process();
        int[] array = new int[n];
        int New;
        do {
            New = 1;
            System.out.println("");
            // In ra mang va tinh tich cac phan tu trong mang
            for (int j = 0; j < n; j++) {
                System.out.print(" " + array[j]);
                New *= array[j];
            }
            int i = n - 1;
            do {
                if (array[i] == 0) {
                    array[i] = 1;
                    for (int j = n - 1; j > i; j--) {
                        array[j] = 0;
                    }
                    break;
                } else
                    i--;
            } while (i >= 0);
        } while (New != 1);
    }
}

File: app/src/at-tamle/java/asiantech/internship/summer/Kotlin/Binary.kt
package asiantech.internship.summer.Kotlin

import java.util.Scanner
object Binary {
    fun process(): Int {
        val input = Scanner(System.`in`)
        var n = 0

                n = input.nextInt()

        return n
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val n = process()
        val array = IntArray(n)
        var New: Int
        do {
            New = 1
            println()

            for (j in 0 until n) {
                print(" " + array[j])
                New *= array[j]
            }
            var i = n - 1
            do {
                if (array[i] == 0) {
                    array[i] = 1
                    for (j in n - 1 downTo i + 1) {
                        array[j] = 0
                    }
                    break
                } else
                    i--
            } while (i >= 0)
        } while (New != 1)
    }
}
